CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc

```diff
@@ -365,6 +544,91 @@ void SiPixelLorentzAnglePCLWorker::dqmAnalyze(edm::Event const& iEvent,
         } else if (subDetID == PixelSubdetector::PixelEndcap) {
           // what to do here?
           edm::LogInfo("SiPixelLorentzAnglePCLWorker") << "do Nothing here?" << std::endl;
+          DetId detIdObj = recHit->geographicalId();
+          const PixelGeomDetUnit* theGeomDet = dynamic_cast<const PixelGeomDetUnit*>(tracker->idToDet(detIdObj));
+          if (!theGeomDet)
+            continue;
+
+          const PixelTopology* topol = &(theGeomDet->specificTopology());
+
+          if (!topol)
+            continue;
+
+          sideF_ = tTopo->pxfSide(detIdObj);
+          diskF_ = tTopo->pxfDisk(detIdObj);
+          bladeF_ = tTopo->pxfBlade(detIdObj);
+          panelF_ = tTopo->pxfPanel(detIdObj);
+          moduleF_ = tTopo->pxfModule(detIdObj);
+
+          const SiPixelRecHit* recHitPix = dynamic_cast<const SiPixelRecHit*>((*recHit).hit());
+          if (!recHitPix)
+            continue;
+          rechitF_.x = recHitPix->localPosition().x();
+          rechitF_.y = recHitPix->localPosition().y();
+          SiPixelRecHit::ClusterRef const& cluster = recHitPix->cluster();
+
+          pixinfo_ = fillPix(*cluster, topol);
+
+          // fill entries in clust_
+            
+          clustF_.x = (cluster)->x();
+          clustF_.y = (cluster)->y();
+          clustF_.charge = (cluster->charge())/1000.;
+          clustF_.size_x = cluster->sizeX();
+          clustF_.size_y = cluster->sizeY();
+          clustF_.maxPixelCol = cluster->maxPixelCol();
+          clustF_.maxPixelRow = cluster->maxPixelRow();
+          clustF_.minPixelCol = cluster->minPixelCol();
+          clustF_.minPixelRow = cluster->minPixelRow();
+        
+
+          // fill the trackhit info
+          TrajectoryStateOnSurface tsos = itTraj->updatedState();
+          if (!tsos.isValid()) {
+            edm::LogWarning("SiPixelLorentzAnglePCLWorker") << "tsos not valid" << std::endl;
+            continue;
+          }
+          LocalVector trackdirection = tsos.localDirection();
+          LocalPoint trackposition = tsos.localPosition();
+
+          if (trackdirection.z() == 0)
+            continue;
+          // the local position and direction
+          trackhitF_.alpha = atan2(trackdirection.z(),trackdirection.x());
+          trackhitF_.beta = atan2(trackdirection.z(),trackdirection.y());
+          trackhitF_.gamma = atan2(trackdirection.x(),trackdirection.y());
+          trackhitF_.x = trackposition.x();
+          trackhitF_.y = trackposition.y();
+            
+          float cotalpha = 1. / TMath::Tan(trackhit_.alpha);
+          float cotbeta = 1. / TMath::Tan(trackhit_.beta);
+          float locBx = 1.;
+          if (cotbeta < 0.)
+             locBx = -1.;
+          float locBz = locBx;
+          if (cotalpha < 0.)
+            locBz = -locBx;
+
+          auto detId = detIdObj.rawId();
+          int TemplID = -9999;
+          TemplID = templateDBobject_->getTemplateID(detId);
```

Comment:
```suggestion
          int TemplID = templateDBobject_->getTemplateID(detId);
```
can't this variable be assigned directly?